izpack-panel/src/main/java/com/izforge/izpack/panels/userinput/UserInputPanel.java

```diff
@@ -686,7 +686,7 @@ private String getCustomSummary(GUICustomField customField)
                 column=1; //Reset to first column
             }
 
-            key = installData.getMessages().get(installData.getMessages().get(labels.get(column-1)));
+            key = userInputModel.getConfig().getLocalString(userInputModel.getConfig().getLocalString(labels.get(column-1)));
```

Comment:
For variables substitution there is a ```VariableSubstitutor``` (```VariableSubstitutorImpl```) that uses the ```VariableSubstitutorReader``` to replace variable values.

For locale strings there is no "variable" replacement available in ```Messages``` (```LocaleDatabase```). If there is a deeper replacement necessary your solution will not work.
Can you provide a simple test case for this issue?

Reply:
Hi, I attached an example in the JIRA - when you click through to summary screen the fields from UserInput screen appear as summary keys rather then the translated strings.

As for this particular line - as far as I can tell replacing it with `userInputModel.getConfig().getLocalString(labels.get(column-1))` works correctly.

Reply:
Please refer to my pull request https://github.com/izpack/izpack/pull/744 in which I have taken care of variable replacements and uploaded test-var.zip file to the ticket which includes variables in the texts.

I think to fix this issue we need to handle the user input langpack in the same way as custom langpack. I have done the same in my pull request, please refer to the same and let me know if I am missing something. In my opinion, there should not have been a separate langpack for user input panel. We could have used same custom langpack which is used for custom panels. It would have avoided this issue. Maybe we can think of deprecating user input langpack and ask the users to use custom langpack instead.

Reply:
@spyrkob/@reinhapa , can you please close this PR as it gets replaced by https://github.com/izpack/izpack/pull/744? Thanks.

Reply:
@spyrkob / @reinhapa , can you please close this PR as it gets replaced by #744? Thanks.